### ISSUE TITLE
Vagrant server is unreachable

### ISSUE THREAD

**Issue by piotrzarzycki21** [Moonshine-IDE/Moonshine-IDE#1256]
On my PC I have several instances of SHI vagrant servers. Unfortunately for some reason Moonshine doesn't see whether server is available. While when I build Moonshine from sources in my IDE server is being discovered correctly. As part of this issue I'm going to investigate what is happening.

**Comment by piotrzarzycki21** [comment]
Here is logs after adding additional logging from `VagrantUtil.checkStates`

Note that nomad12.startcloud.com server is running:

Moonshine build from source locally from my IDE. 
In debug mode there are more information - process reports that two servers are unavailable and one nomad12 is ready:

```
[Vagrant Status] Checking server status: http://restapi.jupiter.startcloud.com:8080/info
[Vagrant Status] Checking server status: http://restapi.venus.startcloud.com:8080/info
[Vagrant Status] Checking server status: http://restapi.nomad12.startcloud.com:8080/info

[Vagrant Status] Server error: 2032 Error #2032: Stream Error. URL: http://restapi.jupiter.startcloud.com:8080/info
[Vagrant Status] Server error: 2032 Error #2032: Stream Error. URL: http://restapi.venus.startcloud.com:8080/info
```

In version 3.3.8 custom build reports that all url are unavailable. There are less information in case of errors.

```
[Vagrant Status] Checking server status: http://restapi.jupiter.startcloud.com:8080/info
[Vagrant Status] Checking server status: http://restapi.venus.startcloud.com:8080/info
[Vagrant Status] Checking server status: http://restapi.nomad12.startcloud.com:8080/info

[Vagrant Status] Server error: 2032 Error #2032
[Vagrant Status] Server error: 2032 Error #2032
[Vagrant Status] Server error: 2032 Error #2032
```

I need to think whether there is any other way to discover availability of servers than using URLLoader. 

Thought:
- What if that part of the code would be moved to Haxe part of the project and we will use Haxe versions of classes ?

**Comment by rat-moonshine** [comment]
> Thought:
> 
 > What if that part of the code would be moved to Haxe part of the project and we will use Haxe versions of classes ?

In my opinion we shall need to do that at some time anyway. But I wonder if you sees any immediate advantage of doing that?

**Comment by piotrzarzycki21** [comment]
Yes - resolving my issue and be able to use that option :) Currently I cannot deploy SHP to my server due to that issue.

**Comment by piotrzarzycki21** [comment]
It was just a thought I don't know if it's helps, but I don't have any other idea. Unless you have ?

**Comment by rat-moonshine** [comment]
> It was just a thought I don't know if it's helps, but I don't have any other idea. Unless you have ?

I can check and get back to you, if you prefer. At this point, I'm not very clear what is the problem with `URLLoader` class that you're looking for an alternative.

**Comment by piotrzarzycki21** [comment]
Neither do I. It's just doesn't work in released application and in IDE does. My attempt of resolving this is rather blind shot. If you reproduce the issue that would be good, but you may as well not reproduce it. I will provide some scenario here so you could try.

**Comment by JoelProminic** [comment]
I cleared all but one of my Linked Instances and restarted Moonshine Development, and I found it detected my Vagrant instances from both Super.Human.Installer and Development:

![image](https://github.com/user-attachments/assets/37ac7f15-d795-4152-80dd-61d316b7a179)

Note that these entries require updates to the hosts file to work.  Similarly, if the IP address has changed since the host entry was added, then you may see inconsistent statuses for the instances (in my screenshot test-1201 is actually running instead of domino-1).

**Comment by rat-moonshine** [comment]
Is @JoelProminic 's insight gives any hint to the problem you faced @piotrzarzycki21 ?

**Comment by rat-moonshine** [comment]
I remember instances when I reported problems with `URLLoader` to @joshtynjala some time to time (for Native/GBAuth). I vaguely remember I required to update Openfl/Lime libraries post Josh's fix, and they were mostly in 'development' branch of the libraries. 

I'm wondering if you're seeing something similar around that API again..
